### PR TITLE
boards: shields: Introduce atmel_rf2xx module shield

### LINKED ISSUE #26259: Add AT86RF233 REB Xplained Pro Extension shield
**Is your feature request related to a problem? Please describe.**
The IEEE 802.15.4 driver is available since Zephyr 2.2.0, however no shield is available at moment.
There are board configuration at samples/net/sockets/echo_(server/client) and the board atsamr21_xpro that is a SIP (SAMD21 + AT86RF233).

**Describe the solution you'd like**
Add AT86RF233 REB Xplained Pro Extension as Shield on boards/shield.

**Describe alternatives you've considered**
The AT86RF233 REB Xplained Pro Extension can be used as reference for any development on Zephyr that may use RF2xx driver.
https://www.microchip.com/DevelopmentTools/ProductDetails/PartNO/ATREB233-XPRO
http://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-42188-REB233-XPRO-User-Guide.pdf

**Additional context**
https://github.com/zephyrproject-rtos/zephyr/pull/25591#pullrequestreview-426283062

CC: @markus-becker-tridonic-com @jfischer-phytec-iot

### PR DESCRIPTION
**Pull request by nandojve**
The Atmel RF2xx module shield is a generic solution to enable any Atmel AT86RF2xx IEEE 802.15.4 transceiver. This module enables IEEE 802.15.4 RF2xx Zephyr driver.

The Atmel RF2xx module shield enables any board with an Atmel Xplained, Xplained-Pro, Arduino or MikroBus expansion header to connect to networks operation with IEEE 802.15.4, OpenThread or any other stack based on this media type.

The Atmel RF2xx module is configured to allow interoperate with other medias like Ethernet. User need configure network stack properly.
    
Fixes #26259.

To allow shield be available, Atmel Xplained and Xplained-Pro header were develop and added to some boards. This allow remove board config at samples/net/sockets/echo_server(client). Tests variations are enabled at samples/net/sockets/echo_server(client) for all shields variations. This allow tracking changes on RF2XX driver to make sure any changes on Zephyr RF2XX driver at least keep code building.

~This uses commits from #26718 to ensures spi communication.~

### CODE REVIEW

**Comment by nandojve** [comment]
CC @markus-becker-tridonic-com @jfischer-phytec-iot @benpicco @nzmichaelh

**Review by jfischer-no — APPROVED** [review]
:+1:

File: dts/bindings/gpio/atmel-xplained-header.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2020 Gerson Fernando Budke <nandojve@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    GPIO pins exposed on Atmel Xplained headers.
+
+    The Xplained layout provide a standard 10 pin header.  A board can have
+    one or more headers and can share pins.  This connector was develop to
```
Comment (nzmichaelh) [inline dts/bindings/gpio/atmel-xplained-header.yaml:8]: 'was developed'

File: dts/bindings/gpio/atmel-xplained-header.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    match with Atmel AVR Xmega devices GPIO port plus power signals.  The Atmel
```
Comment (nzmichaelh) [inline dts/bindings/gpio/atmel-xplained-header.yaml:9]: nit: s/Xmega/XMEGA/

**Review by nzmichaelh — APPROVED** [review]
LGTM with changes.

File: boards/shields/atmel_rf2xx/Kconfig.defconfig [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+config IEEE802154_RF2XX
+	default y
+	depends on IEEE802154
+
+config NET_CONFIG_IEEE802154_DEV_NAME
+	default "RF2XX_0"
+	depends on IEEE802154_RF2XX
+
+orsource "boards/*.defconfig"
```
[the remark this answers is not in the document]
Comment (nandojve) [reply boards/shields/atmel_rf2xx/Kconfig.defconfig:21]: it includes boards/*.defconfig if they exists.

File: boards/shields/atmel_rf2xx/atmel_rf2xx_mikrobus.overlay [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&mikrobus_spi {
+	status = "okay";
+
+	/* CS */
+	cs-gpios = <&mikrobus_header 2
+		    (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
```
[the remark this answers is not in the document]
Comment (nandojve) [reply boards/shields/atmel_rf2xx/atmel_rf2xx_mikrobus.overlay]: Fixed

File: dts/bindings/gpio/atmel-xplained-header.yaml (hunk 1 above)
Comment (nandojve) [reply dts/bindings/gpio/atmel-xplained-header.yaml]: Done.

File: dts/bindings/gpio/atmel-xplained-header.yaml (hunk 2 above)
Comment (nandojve) [reply dts/bindings/gpio/atmel-xplained-header.yaml]: Done.

**Pull request merged**